frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx

```diff
@@ -1,100 +1,322 @@
+  };
+
   return (
-    <Tabs defaultValue={TABS.Overview} data-testid="embedding-preview">
-      <Tabs.List p="0 1.5rem">
-        <Tabs.Tab value={TABS.Overview}>{t`Overview`}</Tabs.Tab>
-        <Tabs.Tab value={TABS.Parameters}>{t`Parameters`}</Tabs.Tab>
-        <Tabs.Tab value={TABS.Appearance}>{t`Appearance`}</Tabs.Tab>
-      </Tabs.List>
-      <Tabs.Panel value={TABS.Overview}>
-        <OverviewSettings
-          embedType={embedType}
-          resource={resource}
-          resourceType={resourceType}
-          iframeUrl={iframeUrl}
-          token={token}
-          siteUrl={siteUrl}
-          secretKey={secretKey}
-          params={params}
-          displayOptions={displayOptions}
-        />
-      </Tabs.Panel>
-      <Tabs.Panel value={TABS.Parameters}>
-        <ParametersSettings
-          activePane={activePane}
-          resource={resource}
-          resourceType={resourceType}
-          resourceParameters={resourceParameters}
-          embeddingParams={embeddingParams}
-          lockedParameters={lockedParameters}
-          parameterValues={parameterValues}
-          embedType={embedType}
-          iframeUrl={iframeUrl}
-          token={token}
-          siteUrl={siteUrl}
-          secretKey={secretKey}
-          params={params}
-          displayOptions={displayOptions}
-          onChangeEmbeddingParameters={onChangeEmbeddingParameters}
-          onChangeParameterValue={onChangeParameterValue}
-          onChangePane={onChangePane}
-        />
-      </Tabs.Panel>
-      <Tabs.Panel value={TABS.Appearance}>
-        <AppearanceSettings
-          activePane={activePane}
-          embedType={embedType}
-          resource={resource}
-          resourceType={resourceType}
-          iframeUrl={iframeUrl}
-          token={token}
-          siteUrl={siteUrl}
-          secretKey={secretKey}
-          params={params}
-          displayOptions={displayOptions}
-          onChangePane={onChangePane}
-          onChangeDisplayOptions={onChangeDisplayOptions}
-        />
-      </Tabs.Panel>
-    </Tabs>
+    <Stack spacing={0}>
+      <EmbedModalContentStatusBar
+        resourceType={resourceType}
+        isPublished={resource.enable_embedding}
+        hasSettingsChanges={hasSettingsChanges}
+        onSave={handleSave}
+        onUnpublish={handleUnpublish}
+        onDiscard={handleDiscard}
+      />
+
+      <Tabs defaultValue={TABS.Overview} data-testid="embedding-preview">
+        <Tabs.List p="0 1.5rem">
+          <Tabs.Tab value={TABS.Overview}>{t`Overview`}</Tabs.Tab>
+          <Tabs.Tab value={TABS.Parameters}>{t`Parameters`}</Tabs.Tab>
+          <Tabs.Tab value={TABS.Appearance}>{t`Appearance`}</Tabs.Tab>
+        </Tabs.List>
+        <Tabs.Panel value={TABS.Overview}>
+          <OverviewSettings
+            resourceType={resourceType}
+            selectedServerCodeOption={selectedServerCodeOption}
+            serverEmbedCodeSlot={
+              <ServerEmbedCodePane
```

Comment:
Can we put this code into a variable/function, something like this?

```

const getEmbedCodePane = (variant: "appearance" | "overview" | "parameters") => (
	<ServerEmbedCodePane
                className="flex-full w-full"
                variant={variant}
                initialPreviewParameters={initialPreviewParameters}
                resource={resource}
                resourceType={resourceType}
                siteUrl={siteUrl}
                secretKey={secretKey}
                params={previewParametersBySlug}
                displayOptions={displayOptions}
                serverCodeOptions={serverCodeOptions}
                selectedServerCodeOptionName={selectedServerCodeOptionName}
                setSelectedServerCodeOptionName={
                  setSelectedServerCodeOptionName
                }
              />
)

```

I ran the `serverEmbedCodeSlot` props into a diff checker and `variant` is the only thing I saw was different

Reply:
or use `TABS.Apperance` etc etc as the parameter value